packages/app/src/cli/services/dev.ts

```diff
@@ -378,6 +414,55 @@ async function devUIExtensionsTarget({
   }
 }
 
+interface DevNonPreviewableExtensionsOptions {
+  app: AppInterface
+  apiKey: string
+  url: string
+  token: string
+  extensions: UIExtension[]
+  remoteExtensions: {
+    [key: string]: string
+  }
+  specifications: UIExtensionSpec[]
+}
+
+export function devNonPreviewableExtensionTarget({
+  extensions,
+  app,
+  url,
+  apiKey,
+  token,
+  remoteExtensions,
+  specifications,
+}: DevNonPreviewableExtensionsOptions) {
+  return {
+    prefix: 'extensions',
+    action: async (stdout: Writable, stderr: Writable, signal: AbortSignal) => {
+      await Promise.all(
+        extensions.map((extension) => {
+          const registrationId = remoteExtensions[extension.localIdentifier]
+          if (!registrationId) throw new AbortError(`Extension ${extension.localIdentifier} not found on remote app.`)
+
+          return Promise.all([
```

Comment:
does it make sense to have nested promise.all?
if you return here an array of promises, can we flatten that at the top level?